debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbPartition.java

```diff
@@ -35,13 +48,34 @@ public MongoDbPartition(String serverId) {
     MongoDbPartition(String serverId, String replicaSetName) {
         this.serverId = serverId;
         this.replicaSetName = replicaSetName;
+        this.multiTaskGen = -1;
+        this.taskId = 0;
+        this.taskCount = 1;
+        this.multiTaskEnabled = false;
+    }
+
+    MongoDbPartition(String serverId, String replicaSetName, boolean multiTaskEnabled, int multiTaskGen, int taskId, int taskCount) {
+        this.serverId = serverId;
+        this.replicaSetName = replicaSetName;
+        this.multiTaskEnabled = multiTaskEnabled;
+        this.multiTaskGen = multiTaskGen;
+        this.taskId = taskId;
+        this.taskCount = taskCount;
     }
 
     @Override
     public Map<String, String> getSourcePartition() {
-        return replicaSetName != null
-                ? Collect.hashMapOf(SERVER_ID_KEY, serverId, REPLICA_SET_NAME, replicaSetName)
-                : Collect.hashMapOf(SERVER_ID_KEY, serverId);
+        Map<String, String> partition = new HashMap<>();
+        if (this.multiTaskEnabled) {
+            partition.put(TASK_ID_KEY, String.valueOf(this.taskId));
+            partition.put(MULTI_TASK_GEN_KEY, String.valueOf(this.multiTaskGen));
+            partition.put(TASK_COUNT_KEY, String.valueOf(taskCount));
```

Comment:
Why do we need to use taks count?

Reply:
We included task count in the partition to guard against accidental misconfiguration. Ex: if generation 0 has 3 tasks, and generation 1 is configured with `prev_tasks=1` the connector will fail to find any offsets and exit instead of silently using the incorrect configuration.